packages/crypto/src/errors.rs

```diff
@@ -84,11 +75,22 @@ impl CryptoError {
     pub fn pubkey_err<S: Into<String>>(msg: S) -> Self {
         CryptoError::PublicKeyErr {
             msg: msg.into(),
-            error_code: 5,
             #[cfg(feature = "backtraces")]
             backtrace: Backtrace::capture(),
         }
     }
+
+    /// Numeric error code that can easily be passed over the
+    /// contract VM boundary.
+    pub fn code(&self) -> u32 {
```

Comment:
I was trying to write a helper like this. It seems you cannot access the internal "fields" (i.e. a putative `error_code`) without going through matching. So, I guess this way makes more sense.

Reply:
Yes, because the type system does not know every case has the same `error_code` property.

But we don't even need to have the data stored in the property when the information is already stored in the enum case. This avoids situations like `PublicKeyErr { error_code: 5 }` and `PublicKeyErr { error_code: 6 } // bug, sorry bro`.